pallets/circuit/src/lib.rs

```diff
@@ -273,31 +346,64 @@ pub mod pallet {
     }
 
     impl<T: Config> OnLocalTrigger<T> for Pallet<T> {
-        fn on_local_trigger(_origin: &OriginFor<T>, _trigger: LocalTrigger<T>) -> DispatchResult {
-            // ToDo: pallet-circuit x-t3rn# : Authorize : Check TriggerAuthRights for local triggers
-
-            // ToDo: pallet-circuit x-t3rn# : Validate : insurance for reversible side effects if necessary
-
-            // ToDo: pallet-circuit x-t3rn# : Charge : fees
-
-            // ToDo: pallet-circuit x-t3rn# : Design Storage - Propose and organise the state of Circuit. Specifically inspect the state updates in between CircuitPortal + Circuit
-
-            // ToDo: pallet-circuit x-t3rn# : Setup : Create new Xtx and modify state - get LocalState (for Xtx) + GlobalState (for Circuit) for exec
-
-            // ToDo: pallet-circuit x-t3rn# : Emit : Connect to CircuitPortal::submit_side_effect_temp( )
+        fn on_local_trigger(origin: OriginFor<T>, trigger: LocalTrigger<T>) -> DispatchResult {
+            // Authorize: Retrieve sender of the transaction.
+            let requester = Self::authorize(origin, CircuitRole::ContractAuthor)?;
 
-            // ToDo: pallet-circuit x-t3rn# : Cancel : Execution on timeout
+            let fresh_or_revoked_exec = match trigger.maybe_xtx_id {
+                Some(_xtx_id) => CircuitStatus::Ready,
+                None => CircuitStatus::Requested,
+            };
+            // Setup: new xtx context
+            let mut local_xtx_ctx: LocalXtxCtx<T> = Self::setup(
+                fresh_or_revoked_exec.clone(),
+                &requester,
+                Zero::zero(),
+                trigger.maybe_xtx_id,
+            )?;
 
-            // ToDo: pallet-circuit x-t3rn# : Apply - Submission : Apply changes to storage after Submit has passed
+            // Charge: Ensure can afford
+            // ToDo: Charge requester for contract with gas_estimation
+            Self::charge(&requester, Zero::zero()).map_err(|_e| {
```

Comment:
Question on this one:

Should this go through the account manager? or should volatile calls stay being charged through this system?

Who gets remunerated for volatile calls?

If so, maybe we need to think about what charging systems should now use the account manager for fees and pay distribution?

Reply:
Also, should this charge post-validation?

Reply:
Awesome! i made a note here, but since i'm not sure what level it should go in, i just left it here https://github.com/orgs/t3rn/projects/1#card-80869132

Reply:
thx!